assets/js/googlesitekit/widgets/datastore/areas.test.js

```diff
@@ -0,0 +1,333 @@
+		unsubscribeFromAll( registry );
+	} );
+
+	describe( 'actions', () => {
+		describe( 'assignWidgetArea', () => {
+			it( 'should assign a registered widget area to a context', () => {
+				// Register the widget area.
+				const slug = 'header';
+				const settings = {
+					priority: 10,
+					title: 'Your Site',
+					subtitle: 'Learn about your site!',
+					icon: '/wp-content/plugins/googlesitekit/header.svg',
+					style: 'boxes',
+				};
+				registry.dispatch( STORE_NAME ).registerWidgetArea( slug, settings );
+
+				// Assign this widget area to the dashboard context.
+				registry.dispatch( STORE_NAME ).assignWidgetArea( slug, 'dashboard' );
+
+				// Get all assigned widget areas for the dashboard context.
+				const dashboardAreas = registry.select( STORE_NAME ).getWidgetAreas( 'dashboard' );
+
+				expect( dashboardAreas ).toHaveLength( 1 );
+				expect( dashboardAreas.some( ( area ) => area.slug === slug ) ).toEqual( true );
+			} );
+		} );
+
+		describe( 'registerWidgetArea', () => {
+			it( 'should register a widget area', () => {
+				const slug = 'header';
+				const settings = {
+					priority: 10,
+					title: 'Your Site',
+					subtitle: 'Learn about your site!',
+					icon: '/wp-content/plugins/googlesitekit/header.svg',
+					style: 'boxes',
+				};
+				registry.dispatch( STORE_NAME ).registerWidgetArea( slug, settings );
+				const state = store.getState();
+
+				expect( registry.select( STORE_NAME ).isWidgetAreaRegistered( slug ) ).toEqual( true );
+				// There is no selector for unassigned widget areas, so we inspect the store directly for
+				// this test.
+				expect( state.areas ).toMatchObject( {
+					[ slug ]: { ...settings, slug },
+				} );
+			} );
+
+			it( 'requires an areaSlug', () => {
+				expect( () => {
+					registry.dispatch( STORE_NAME ).registerWidgetArea( null, {} );
+				} ).toThrow( 'areaSlug is required.' );
+			} );
+
+			it( 'requires settings', () => {
+				expect( () => {
+					registry.dispatch( STORE_NAME ).registerWidgetArea( 'my-cool-slug' );
+				} ).toThrow( 'settings is required.' );
+			} );
+
+			it( 'requires a title, subtitle, and style in settings', () => {
+				expect( () => {
+					registry.dispatch( STORE_NAME ).registerWidgetArea( 'header', {} );
+				} ).toThrow( 'settings.title is required.' );
+
+				expect( () => {
+					registry.dispatch( STORE_NAME ).registerWidgetArea( 'header', {
+						title: 'Analytics Header',
+					} );
+				} ).toThrow( 'settings.subtitle is required.' );
+
+				expect( () => {
+					registry.dispatch( STORE_NAME ).registerWidgetArea( 'header', {
+						title: 'Analytics Header',
+						subtitle: 'Analytics tell you about visitors',
+					} );
+				} ).toThrow( 'settings.style is required.' );
+
+				expect( () => {
+					registry.dispatch( STORE_NAME ).registerWidgetArea( 'header', {
+						title: 'Analytics Header',
+						subtitle: 'Analytics tell you about visitors',
+						style: 'composite',
+					} );
+				} ).not.toThrow();
```

Comment:
The `style` tests here can be removed, this should be optional. We could instead test that it receives `boxes` as default value as expected.